scripts/missions/amk/13_A_Challenge_You_Could_Be_a_Winner.lua

```diff
@@ -17,28 +19,353 @@ mission.reward =
+            not player:hasKeyItem(xi.ki.TRIVIA_CHALLENGE_KUPON) and
+            player:needToZone() == false
+        end,
+
+        [xi.zone.BEAUCEDINE_GLACIER] =
+        {
+            ['Lonely_Evergreen'] =
+            {
+                onTrigger = function(player, npc)
+                    return mission:progressEvent(504)
+                end,
+            },
+
+            onEventFinish =
+            {
+                [504] = function(player, csid, option, npc)
+                    if option == 1 then
+                        player:needToZone(true)
+                    end
+                end,
+            }
+        },
+    },
+
+    -- Part 2-b: Beaucedine Glacier
+    {
+        check = function(player, currentMission, missionStatus, vars)
+            return currentMission >= mission.missionId and
+            player:getCharVar('Mission[10][12]progress') == 1 and
+            not player:hasKeyItem(xi.ki.POCKET_MOGBOMB) and
+            not player:hasKeyItem(xi.ki.TRIVIA_CHALLENGE_KUPON) and
+            player:needToZone() == true
+        end,
+
+        [xi.zone.BEAUCEDINE_GLACIER] =
+        {
+            ['Lonely_Evergreen'] =
+            {
+                onTrigger = function(player, npc)
+                    return mission:progressEvent(503)
+                end,
+            },
+
+            ['Goblin_Grenadier'] =
+            {
+                onTrigger = function(player, npc)
+                    local answer = player:getLocalVar('Mission[10][12]pipSet') - 1
+
+                    if answer < 0 then
+                        return mission:progressEvent(508, xi.ki.MAP_OF_THE_NORTHLANDS_AREA)
+                    else
+                        local today = VanadielDayOfTheWeek()
+                        local tomorrow = (today + 1) % 8
+                        local hintsUsed = player:getLocalVar('Mission[10][12]hintsUsed')
+
+                        player:messageSpecial(zones[player:getZoneID()].text.GRENADIER_DAY_HINT, today, tomorrow)
+
+                        return mission:progressEvent(
+                            507,
+                            hintsUsed + 2,
+                            answer,
+                            xi.ki.MAP_OF_THE_NORTHLANDS_AREA,
+                            xi.ki.POCKET_MOGBOMB,
+                            xi.ki.MAP_OF_THE_NORTHLANDS_AREA
+                        )
+                    end
+                end,
+            },
+
+            ['Northwestern_Pip'] =
+            {
+                onTrigger = function(player, npc)
+                    local pipSet = player:getLocalVar('Mission[10][12]pipSet') - 1
+                    local pos = npc:getPos()
+                    local element = xi.amk.helpers.pipSets[pipSet][1]
+
+                    return mission:progressEvent(
+                        510,
+                        pos.x * 1000,
+                        pos.z * 1000,
+                        pos.y * 1000,
```

Comment:
I don't think I've ever seen pos units put into an event like this, interesting!